JujuUtils/JujuUtils.psm1

```diff
@@ -0,0 +1,549 @@
+
+    WARNING: Do not use this to convert large files, as it reads the entire contents of a file
+    into memory. This function may be useful to transfer small amounts of data over a relation
+    without having to worry about encoding or escaping, preserving at the same time any
+    binary info/special
+    characters.
+    .PARAMETER File
+    The path to the file you want to convert. It works for any type of file. Take great care not to
+    try and convert large files.
+    #>
+    [CmdletBinding()]
+    Param (
+        [parameter(Mandatory=$true)]
+        [string]$File,
+        [switch]$Force
+    )
+    PROCESS {
+        if(!(Test-Path $File)) {
+            Throw "No such file: $File"
+        }
+        $f = (Get-Item $File)
+        if($f.Length -gt 1MB -and !$Force) {
+            Throw "File is too big to convert (> 1MB). Use -Force to do it anyway..."
+        }
+        $ct = [System.IO.File]::ReadAllBytes($File)
+        $b64 = [Convert]::ToBase64String($ct)
+        return $b64
+    }
+}
+
+function Write-FileFromBase64 {
+    <#
+    .SYNOPSIS
+    Helper function that converts base64 to bytes and then writes that stream to a file.
+    .PARAMETER File
+    Destination file to write to.
+    .PARAMETER Content
+    Base64 encoded string
+    #>
+    [CmdletBinding()]
+    Param (
+        [Parameter(Mandatory=$true)]
+        [string]$File,
+        [Parameter(Mandatory=$true, ValueFromPipeline=$true)]
+        [string]$Content
+    )
+    PROCESS {
+        $bytes = [Convert]::FromBase64String($Content)
+        [System.IO.File]::WriteAllBytes($File, $bytes)
+    }
+}
+
+function ConvertTo-Base64 {
+    <#
+    .SYNOPSIS
+    Convert string to its base64 representation
+    .PARAMETER Content
+    String to be converted to base64
+    #>
+    [CmdletBinding()]
+    Param (
+        [Parameter(Mandatory=$true, ValueFromPipeline=$true)]
+        [string]$Content
+    )
+    PROCESS {
+        $x = [System.Convert]::ToBase64String([System.Text.Encoding]::Unicode.GetBytes($Content))
+        return $x
+    }
+}
+
+function ConvertFrom-Base64 {
+    <#
+    .SYNOPSIS
+    Convert base64 back to string
+    .PARAMETER Content
+    Base64 encoded string
+    #>
+    [CmdletBinding()]
+    Param (
+        [Parameter(Mandatory=$true, ValueFromPipeline=$true)]
+        [string]$Content
+    )
+    PROCESS {
+        $x = [System.Text.Encoding]::Unicode.GetString([System.Convert]::FromBase64String($content))
+        return $x
+    }
+}
+
+function Get-EncryptedString {
+    <#
+    .SYNOPSIS
+    This is just a helper function that converts a plain string to a secure string and returns the encrypted
+    string representation.
+    .PARAMETER Content
+    The string you want to encrypt
+    #>
+    [CmdletBinding()]
+    Param (
+        [Parameter(Mandatory=$true, ValueFromPipeline=$true)]
+        [string]$Content
+    )
+    PROCESS {
+        $ret = ConvertTo-SecureString -AsPlainText -Force $Content | ConvertFrom-SecureString
```

Comment:
This doesn't work for me when I want to create a New-Object PSCredential($user, $encriptedString)
This function returns a string converted FROM SecureString. I had to remove the pipeing in order to use this.